xcodeproj/internal/xcode_schemes.bzl

```diff
@@ -147,6 +160,13 @@ def _unfocus_schemes(schemes, unfocused_targets):
 
     return focused_schemes
 
+def _pre_post_action(name, target, script_contents):
```

Comment:
We should ensure that the only thing required is `script_contents`. `name` can default to `"Run Script"`: https://github.com/tuist/XcodeProj/blob/c36d3a8501588e280f4199b00c5cfc5a23b27dbf/Sources/XcodeProj/Scheme/XCScheme%2BExecutionAction.swift#L16

Reply:
With that said, we should require `target` (don't set a default), but allow `None`. This at least encourages people to set it, as it's most likely needed for environment variables to be as expected.

Reply:
I was going to leave a comment on the lack of Shell option, but it looks like from that link it's not currently supported by Tuist?

Reply:
@maxwellE Is that something you can add to `XcodeProj`?

Reply:
Yes I can look into that

Reply:
Seems someone was thinking of the same thing? https://github.com/tuist/XcodeProj/pull/721#pullrequestreview-1098649282

Reply:
Now that https://github.com/buildbuddy-io/rules_xcodeproj/commit/7e2fab47ceebbce92771ff17e7a0011efa491eab is merged, we can add `shell` here as well.

Reply:
@brentleyjones In another review comment, I suggest renaming `target` to `execution_context`. Does that sound reasonable to you?

Reply:
lol holy moly